From f8f223da46716a0a888ef5a5a0840fb69dbc7067 Mon Sep 17 00:00:00 2001
From: Julian Sparber <julian@sparber.net>
Date: Wed, 7 Oct 2020 10:36:50 +0200
Subject: [PATCH 061/124] main-window: Block forward navigation when viewer is
 empty

Empty viewer includes: - no selected conversation
                       - no conversation in folder
                       - selected more then one conversation
---
 src/client/application/application-main-window.vala | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/client/application/application-main-window.vala b/src/client/application/application-main-window.vala
index e1e55d0e..3b7c29e7 100644
--- a/src/client/application/application-main-window.vala
+++ b/src/client/application/application-main-window.vala
@@ -1811,8 +1811,11 @@ public class Application.MainWindow :
                     conversations_leaflet.navigate(Hdy.NavigationDirection.FORWARD);
                     focus = this.conversation_list_view;
                 } else {
-                    main_leaflet.navigate(Hdy.NavigationDirection.FORWARD);
-                    focus = this.conversation_viewer.visible_child;
+                    if (this.main_toolbar.selected_conversations == 1 &&
+                        this.selected_folder.properties.email_total > 0) {
+                        main_leaflet.navigate(Hdy.NavigationDirection.FORWARD);
+                        focus = this.conversation_viewer.visible_child;
+                    }
                 }
             }
         } else if (focus != null) {
-- 
2.29.2

